tests/test_split_matrix.py

```diff
@@ -237,3 +239,15 @@ def test_init_from_1d():
 
     res = SplitMatrix([m1, m2])
     assert res.shape == (10, 3)
+
+
+@pytest.mark.parametrize("n_rows", [5, 10, 25])
+def test_matvec(n_rows):
+    np.random.seed(1234)
+    n_cols = 2
+    categories = [f"cat_{val}" for val in range(5)]
+    X = pd.DataFrame(np.random.choice(categories, size=(n_rows, n_cols))).astype(
+        "category"
+    )
+    mat = from_pandas(X)
```

Comment:
This will force the `SplitMatrix` to contain two `CategoricalMatrix`. The default parameters of `from_pandas` were tuned for performance. In this case, if a categorical column contains fewer than 4 categories, it will be transformed into a `SparseMatrix`. Then, again for performance reasons, a `SparseMatrix` column with a density of more than 10% will be transformed to a `DenseMatrix`.
```suggestion
    mat = from_pandas(X, cat_threshold=0)
```